### PR TITLE
Split #3054: Refactoring of OBDScanner

### PR DESCRIPTION
**Pull request by polybassa**

### CODE REVIEW

File: scapy/contrib/automotive/obd/scanner.py [hunk 1]
```diff
@@ -122,139 +91,135 @@ def print_payload(resp):
...
+    _description = "Available DTCs in OBD service 07"
 
-    @staticmethod
-    def get_table_entry(tup):
-        _, _, res = tup
-        label = OBD_Enumerator.get_label(
-            res,
-            positive_case=lambda: OBD_DTC_Enumerator.print_payload(res))
-        return "Service 07", "%d DTCs" % res.count, label
+    def _get_initial_requests(self, **kwargs):
```
Comment (gpotter2) [inline scapy/contrib/automotive/obd/scanner.py:111]: Could you type this function? It returns a mix of Iterables

**Comment by polybassa** [comment]
Sorry, I forgot to enable typing. This was not intended. Now it should be fixed.

**Comment by polybassa** [comment]
ready for merge

**Comment by polybassa** [comment]
@gpotter2 Can this PR get merged?

**Review by gpotter2 — APPROVED** [review]

**Pull request merged**